### ISSUE TITLE
Get website to work from links in plasmapy.org domain

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/plasmapy.github.io#5]
A while back I reserved the plasmapy.org domain for us on [register.com](https://www.register.com/).  I've been trying to set up http://www.plasmapy.org/ so that it points to our page on GitHub which was [https://plasmapy.github.io](https://plasmapy.github.io).  It's been a headache trying to figure out what acronyms like DNS and CNAME mean, but at the moment it appears that at least http://www.plasmapy.org/ is working.

We most importantly want to make sure the following two links work:
[ ] **http://www.plasmapy.org/**  (which should probably be our official link)
[ ] www.plasmapy.org

It would also be good if we could get the following links to redirect to the official link:
[ ] http://plasmapy.org/
[ ] https://www.plasmapy.org/
[ ] https://plasmapy.org/
[ ] https://plasmapy.github.io/
[ ] http://plasmapy.github.io/

**Comment by StanczakDominik** [comment]
As it stands,
 - http://plasmapy.org/ DNS errors
 - https://www.plasmapy.org/ brings up a Not Secure popup
 - https://plasmapy.org/ DNS errors

We could really use some help from web-fluent people on this one!

**Comment by StanczakDominik** [comment]
Hey @sinoroc, here's an email I just got after mergign #15 : 

> The page build completed successfully, but returned the following warning for the `master` branch:
> 
> The CNAME `plasmapy.org` is already taken. Check out https://help.github.com/articles/troubleshooting-custom-domains/#cname-already-taken for more information.
> 
> For information on troubleshooting Jekyll see:
> 
>   https://help.github.com/articles/troubleshooting-jekyll-builds
> 
> If you have any questions you can contact us by replying to this email.

Do you perhaps have any ideas on what that might mean? Perhaps this worked too well and it's working on your fork as well?

**Comment by sinoroc** [comment]
Yes, I received a similar message when I first started with my fork, so that is most likely the issue. I will remove this from my fork.

Aside:
That is really surprising and I wonder if it could not be abused in some way. Seems like a really weak system...

**Comment by StanczakDominik** [comment]
Yeah, it does. That's pretty damn weird, to be honest.

**Comment by sinoroc** [comment]
I am not so sure about the actual issue, but here is what I could gather...

A good way to debug this seems to check the output of following command:
```
$ dig plasmapy.org +nostats +nocmd +nocomments
```

Right now the output looks like this:
```
$ dig plasmapy.org +nostats +nocmd +nocomments

; <<>> DiG 9.10.3-P4-Ubuntu <<>> plasmapy.org +nostats +nocmd +nocomments
;; global options: +cmd
;plasmapy.org.                  IN      A
plasmapy.org.           55      IN      SOA     dns102.a.register.com. root.register.com. 2016062320 28800 7200 604800 14400
```

And that does not seem to point anywhere near GitHub's servers :/

**Comment by sinoroc** [comment]
On the other hand:
```
$ dig www.plasmapy.org +nostats +nocmd +nocomments

; <<>> DiG 9.10.3-P4-Ubuntu <<>> www.plasmapy.org +nostats +nocmd +nocomments
;; global options: +cmd
;www.plasmapy.org.              IN      A
www.plasmapy.org.       1871    IN      A       192.30.252.153
www.plasmapy.org.       1871    IN      A       192.30.252.154
```
That looks like it points to GitHub.

To have both `plasmapy.org` and `www.plasmapy.org`, the DNS configuration should look somewhat like this:
```
A      @    192.30.252.154
A      @    192.30.252.153
CNAME  www  plasmapy.github.io
```

**Comment by StanczakDominik** [comment]
Hey @namurphy were you doing anything with DNS settings for the website? Does the `dig` command give you anything to work with?

**Comment by namurphy** [comment]
I haven't done anything with DNS settings for the website in a long time (except that I just set [docs.plasmapy.org](http://docs.plasmapy.org) forward to our read the docs page a few minutes ago).

I just tried to get to [http://www.plasmapy.org](http://www.plasmapy.org) and the page was missing.  Hm...

**Comment by sinoroc** [comment]
Right now GitHub redirects http://www.plasmapy.org/ to http://plasmapy.org/
But http://plasmapy.org/ is not configured in the DNS.

My recommendation is to set the DNS configuration to something like this:
```
A      @    192.30.252.154
A      @    192.30.252.153
CNAME  www  plasmapy.github.io
```

This seems like the best way to have both http://www.plasmapy.org/ and http://plasmapy.org/ working.

**Comment by sinoroc** [comment]
```
$ curl -I http://www.plasmapy.org
HTTP/1.1 301 Moved Permanently
Server: GitHub.com
Date: Thu, 25 Jan 2018 22:49:06 GMT
Content-Type: text/html
Content-Length: 178
Location: http://plasmapy.org/
```

Which is perfectly fine. Unless you want to have http://plasmapy.org as the "canonical" URL, in which case I guess we only need to change the `fqdn` in the `.travis.yml`.

**Comment by StanczakDominik** [comment]
@namurphy I think you fixed most of it! Only https left to go. That probably involves getting a certificate somehow.

**Comment by sinoroc** [comment]
That's a great step.
The GitHub help pages recommended to use the `www` subdomain:
https://help.github.com/articles/about-supported-custom-domains/#www-subdomains

> We strongly recommend that you use a www subdomain for these reasons:
> * It gives your GitHub Pages site the benefit of our Content Delivery Network.
> * It is more stable because it is not affected by changes to the IP addresses of GitHub's servers.
> * Pages will load significantly faster because Denial of Service attack protection can be implemented more efficiently.

From what I understand, it would be only a matter of changing the `fqdn` value in `.travis.yml`.

**Comment by sinoroc** [comment]
Having SSL / https on the custom domain is a related but different problem.
A popular solution is to use _CloudFlare_.
https://blog.cloudflare.com/secure-and-fast-github-pages-with-cloudflare/

**Comment by namurphy** [comment]
Thank you again, @sinoroc!  I just changed `fqdn` to `www.plasmapy.org` in `.travis.yml`.

**Comment by mathagician** [comment]
the issue is no longer observed. @namurphy can u please close it if that is so or else mention what is pending.
